src/pybind/mgr/dashboard/services/rgw_client.py

```diff
@@ -1183,6 +1183,74 @@ def create_topic(self, request=None, name: str = '',
             raise DashboardException(msg=str(e), component='rgw')
         return result
 
+    @RestClient.api_put('/{bucket_name}?notification')
+    def set_notification(self, bucket_name, notification, request=None):
+        # pylint: disable=unused-argument
+        try:
+            notification = notification.strip()
+
+            if notification.startswith('{'):
+                notification = self.dict_to_xml(notification)
+
+            if not notification.startswith('<NotificationConfiguration'):
+                notification = (
+                    f'<NotificationConfiguration>{notification}</NotificationConfiguration>'
+                )
+
+            result = request(data=notification)  # type: ignore
+
+        except RequestException as e:
+            raise DashboardException(msg=str(e), component='rgw')
+
+        return result
+
+    @RestClient.api_get('/{bucket_name}?notification')
+    def get_notification(self, bucket_name, request=None):
+        # pylint: disable=unused-argument
+        try:
+            result = request(
+                raw_content=True,
+                headers={'Accept': 'text/xml'}
+            ).decode()  # type: ignore
+
+            notification_config = xmltodict.parse(result)
+            notification_configuration = notification_config.get('NotificationConfiguration', {})
```

Comment:
just making the name of the variable what it means..
```suggestion
            notification_config_dict = xmltodict.parse(result)
            notification_configuration = notification_config_dict.get('NotificationConfiguration', {})
```